bin/saya/src/args/shard.rs

```diff
@@ -0,0 +1,69 @@
+use std::fmt::Display;
+use std::str::FromStr;
+
+use clap::builder::PossibleValue;
+use clap::{Args, ValueEnum};
+use katana_primitives::FieldElement;
+use saya_core::SayaMode;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct SayaModeArg(pub SayaMode);
+
+#[derive(Debug, Args, Clone)]
+pub struct SettlementOptions {
+    #[arg(help = "Choose either ephemeral or persistent saya mode.")]
+    #[arg(long = "mode")]
+    pub saya_mode: SayaModeArg,
+
+    #[arg(help = "The settlement contract address.")]
+    #[arg(long)]
+    pub settlement_contract: FieldElement,
```

Comment:
This one may be an option, as the sharded execution is not dependent on a settlement contract, but only the world's address.

We should also ensure the settlement contract is not none when run in persistent mode.

Reply:
done